### DISCUSSION TITLE
Is there any way to read environment variables without loader?

### DISCUSSION THREAD

**Discussion by damyo** [remix-run/remix#4335] [Q&A]
Hi,
I am using supabase and I have saved the url value and anon key value in `.dev.vars` file (because I am using cloudflare pages).

And this environment variable seems to be accessible only by the loader. So I am using it like this.

```js
export const loader = async ({ context }) => {
  return {
    url: context.SUPABASE_URL,
    key: context.SUPABASE_ANON_KEY
  }
}

export default function App() {
  const data = useLoaderData()
  const supabase = createClient(data.url, data.key)
  ...
}
```

The problem is that I have to repeatedly write code like the above in many pages.

```js
export const supabase = createClient(process.env.SUPABASE_URL, process.env.SUPABASE_ANON_KEY)
```

I want to import and use it after declaring it only once as above. I also considered using provider, but it failed because useContext is not available inside loader. I've been googling for a few days and still can't find a way. Is there no way?

**Comment by sergiodxa** [comment]
In Cloudflare there’s no other way to access env variables since they’re not global, but maybe you could do `globalThis.env = yourEnvFromLoader`
